src/main/java/com/cloudbees/hudson/plugins/folder/computed/DefaultOrphanedItemStrategy.java

```diff
@@ -273,19 +259,35 @@ private static boolean disabled(TopLevelItem item) {
             Queue<Run<?, ?>> abortedBuilds = new LinkedList<>();
             for (I item: orphaned) {
                 for (Job<?, ?> job: item.getAllJobs()) {
+                    if (job instanceof hudson.model.Queue.Task) {
+                        hudson.model.Queue jenkinsQueue = Jenkins.get().getQueue();
+                        hudson.model.Queue.Task task = (hudson.model.Queue.Task) job;
+                        for (hudson.model.Queue.Item pendingBuild: jenkinsQueue.getItems(task)) {
+                            jenkinsQueue.cancel(pendingBuild);
```

Comment:
FYI possibly already covered by https://github.com/jenkinsci/jenkins/blob/1a42044347502c59cd9c39d1be67f654df02908a/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java#L486

Reply:
I was going to use this at first, but then I noticed `Queue#cancel(Task)` only cancel the first item of the list:
```java
    /**
     * Cancels the item in the queue. If the item is scheduled more than once, cancels the first occurrence.
     *
     * @return true if the project was indeed in the queue and was removed.
     *         false if this was no-op.
     */
    public boolean cancel(Task p) {
        lock.lock();
        try { try {
            LOGGER.log(Level.FINE, "Cancelling {0}", p);
            for (WaitingItem item : waitingList) {
                if (item.task.equals(p)) {
                    return item.cancel(this);
                }
            }
            // use bitwise-OR to make sure that both branches get evaluated all the time
            return blockedProjects.cancel(p) != null | buildables.cancel(p) != null;
        } finally { updateSnapshot(); } } finally {
            lock.unlock();
        }
    }
```
Not being acquainted to the Jenkins model, I chose the safety by cancelling the list of items.

Reply:
Probably safest to redundantly cancel all items as you do here. In practice it would rarely matter since queue items would be coalesced; they would not be when using parameterized builds, but of course branch projects are much less likely to be parameterized.